src/Analyzers/CSharp/Analyzers/UseCollectionExpression/AbstractCSharpUseCollectionExpressionDiagnosticAnalyzer.cs

```diff
@@ -63,6 +63,7 @@ protected sealed override void InitializeWorker(AnalysisContext context)
             // diagnostics with location outside this node. Without the containing CodeBlockStartAction, our reported
             // diagnostic would be classified as a non-local diagnostic and would not participate in lightbulb for
             // computing code fixes.
-            context.RegisterCodeBlockStartAction<SyntaxKind>(InitializeWorker);
+            var expressionType = compilation.ExpressionOfTType();
```

Comment:
the majority of this PR is just getting this type and passing it along.

Reply:
note: it is computed up front and passed along since actually requesting types from the compiler is not fast, and doing it for each expression we check would actually be expensive.  it is worthwhile to retrieve once and pass along given the number of checks we may be doing.